events/parallel.go

```diff
@@ -0,0 +1,89 @@
+package events
+
+import (
+	"context"
+	"sync"
+)
+
+type ConsumerPool struct {
+	maxConcurrency int
+	maxQueue       int
+
+	do func(*RepoAppend)
+
+	feeder chan *consumerTask
+
+	lk     sync.Mutex
+	active map[string][]*consumerTask
```

Comment:
What is the key for? To keep separate lists per PDS being consumed from? I would have expected something like `map[util.Uid][]int64`, that is, a list of outstanding seq from all the PDSes being consumed from.

Reply:
The key is the did of the user, to ensure we dont process events for a given repo in parallel